<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
 <HEAD>
   <TITLE> [PATCH 1/5] Always build wayland-scanner
   </TITLE>
   <LINK REL="Index" HREF="index.html" >
   <LINK REL="made" HREF="mailto:wayland-devel%40lists.freedesktop.org?Subject=Re%3A%20%5BPATCH%201/5%5D%20Always%20build%20wayland-scanner&In-Reply-To=%3C1400512363-30116-1-git-send-email-thierry.reding%40gmail.com%3E">
   <META NAME="robots" CONTENT="index,nofollow">
   <style type="text/css">
       pre {
           white-space: pre-wrap;       /* css-2.1, curent FF, Opera, Safari */
           }
   </style>
   <META http-equiv="Content-Type" content="text/html; charset=us-ascii">
   <LINK REL="Previous"  HREF="014887.html">
   <LINK REL="Next"  HREF="014893.html">
 </HEAD>
 <BODY BGCOLOR="#ffffff">
   <H1>[PATCH 1/5] Always build wayland-scanner</H1>
    <B>Thierry Reding</B> 
    <A HREF="mailto:wayland-devel%40lists.freedesktop.org?Subject=Re%3A%20%5BPATCH%201/5%5D%20Always%20build%20wayland-scanner&In-Reply-To=%3C1400512363-30116-1-git-send-email-thierry.reding%40gmail.com%3E"
       TITLE="[PATCH 1/5] Always build wayland-scanner">thierry.reding at gmail.com
       </A><BR>
    <I>Mon May 19 08:12:39 PDT 2014</I>
    <P><UL>
        <LI>Previous message: <A HREF="014887.html">[PATCH v4] configure.ac: Change in build system to use the path from pkg-config for wayland-scanner.
</A></li>
        <LI>Next message: <A HREF="014893.html">[PATCH 2/5] Remove wayland_protocoldir definition
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#14892">[ date ]</a>
              <a href="thread.html#14892">[ thread ]</a>
              <a href="subject.html#14892">[ subject ]</a>
              <a href="author.html#14892">[ author ]</a>
         </LI>
       </UL>
    <HR>  
<!--beginarticle-->
<PRE>From: Thierry Reding &lt;<A HREF="http://lists.freedesktop.org/mailman/listinfo/wayland-devel">treding at nvidia.com</A>&gt;

When cross-compiling, wayland-scanner cannot be used to generate files
because it is built for the host architecture. To overcome this, users
currently need to instruct the build not to create wayland-scanner and
provide a native variant in the PATH. This has the disadvantage of not
installing the scanner either and in turn users running a distribution
that was cross-built cannot use it.

This commit changes the build process to always create and install the
wayland-scanner binary. The configure script determines whether to use
the version being built or a native version found in the PATH.

Signed-off-by: Thierry Reding &lt;<A HREF="http://lists.freedesktop.org/mailman/listinfo/wayland-devel">treding at nvidia.com</A>&gt;
---
 Makefile.am  | 11 +++--------
 configure.ac | 31 ++++++++++++++++---------------
 2 files changed, 19 insertions(+), 23 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index c15d8b8b853d..028d0d4837b5 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -61,25 +61,20 @@ nodist_libwayland_client_la_SOURCES =		\
 
 pkgconfig_DATA += src/wayland-client.pc src/wayland-server.pc
 
-if ENABLE_SCANNER
-wayland_scanner = $(top_builddir)/wayland-scanner
 bin_PROGRAMS = wayland-scanner
 wayland_scanner_SOURCES = src/scanner.c
 wayland_scanner_LDADD = $(EXPAT_LIBS) libwayland-util.la
 $(BUILT_SOURCES) : wayland-scanner
 pkgconfig_DATA += src/wayland-scanner.pc
-else
-wayland_scanner = wayland-scanner
-endif
 
 protocol/%-protocol.c : $(top_srcdir)/protocol/%.xml
-	$(AM_V_GEN)$(MKDIR_P) $(dir $@) &amp;&amp; $(wayland_scanner) code &lt; $&lt; &gt; $@
+	$(AM_V_GEN)$(MKDIR_P) $(dir $@) &amp;&amp; $(WAYLAND_SCANNER) code &lt; $&lt; &gt; $@
 
 protocol/%-server-protocol.h : $(top_srcdir)/protocol/%.xml
-	$(AM_V_GEN)$(MKDIR_P) $(dir $@) &amp;&amp; $(wayland_scanner) server-header &lt; $&lt; &gt; $@
+	$(AM_V_GEN)$(MKDIR_P) $(dir $@) &amp;&amp; $(WAYLAND_SCANNER) server-header &lt; $&lt; &gt; $@
 
 protocol/%-client-protocol.h : $(top_srcdir)/protocol/%.xml
-	$(AM_V_GEN)$(MKDIR_P) $(dir $@) &amp;&amp; $(wayland_scanner) client-header &lt; $&lt; &gt; $@
+	$(AM_V_GEN)$(MKDIR_P) $(dir $@) &amp;&amp; $(WAYLAND_SCANNER) client-header &lt; $&lt; &gt; $@
 
 BUILT_SOURCES =					\
 	$(nodist_libwayland_server_la_SOURCES)	\
diff --git a/configure.ac b/configure.ac
index 393310cd6699..d14142bbc8c3 100644
--- a/configure.ac
+++ b/configure.ac
@@ -31,6 +31,17 @@ AC_PROG_CC
 LT_PREREQ([2.2])
 LT_INIT
 
+if test &quot;x$cross_compiling&quot; = &quot;xyes&quot;; then
+	AC_PATH_PROG(WAYLAND_SCANNER, wayland-scanner)
+
+	if test &quot;x$WAYLAND_SCANNER&quot; = &quot;x&quot;; then
+		AC_MSG_ERROR([Cross-compiling but wayland-scanner not found.])
+	fi
+else
+	WAYLAND_SCANNER='$(top_builddir)/wayland-scanner'
+	AC_SUBST([WAYLAND_SCANNER])
+fi
+
 PKG_PROG_PKG_CONFIG()
 PKG_CHECK_MODULES(FFI, [libffi])
 
@@ -52,20 +63,12 @@ AC_CHECK_DECL(CLOCK_MONOTONIC,[],
 	      [[#include &lt;time.h&gt;]])
 AC_CHECK_HEADERS([execinfo.h])
 
-AC_ARG_ENABLE([scanner],
-              [AC_HELP_STRING([--disable-scanner],
-                              [Disable compilation of wayland-scanner])],
-              [],
-              [enable_scanner=yes])
-
 AC_ARG_ENABLE([documentation],
 	      [AC_HELP_STRING([--disable-documentation],
 		              [Disable building the documentation])],
 	      [],
 	      [enable_documentation=yes])
 
-AM_CONDITIONAL(ENABLE_SCANNER, test &quot;x$enable_scanner&quot; = xyes)
-
 AC_ARG_WITH(icondir, [  --with-icondir=&lt;dir&gt;    Look for cursor icons here],
 		     [  ICONDIR=$withval],
 		     [  ICONDIR=${datadir}/icons])
@@ -76,13 +79,11 @@ AC_ARG_WITH(expat, [  --with-expat=&lt;dir&gt;      Use expat from here],
 		   [ expat=$withval
 		     CPPFLAGS=&quot;$CPPFLAGS -I$withval/include&quot;
 		     LDFLAGS=&quot;$LDFLAGS -L$withval/lib&quot; ] )
-if test &quot;x$enable_scanner&quot; = &quot;xyes&quot;; then
-	AC_CHECK_HEADERS(expat.h, [AC_DEFINE(HAVE_EXPAT_H)],
-			 [AC_MSG_ERROR([Can't find expat.h. Please install expat.])])
-	AC_CHECK_LIB(expat, XML_ParserCreate, [EXPAT_LIBS=&quot;-lexpat&quot;],
-		     [AC_MSG_ERROR([Can't find expat library. Please install expat.])])
-	AC_SUBST(EXPAT_LIBS)
-fi
+AC_CHECK_HEADERS(expat.h, [AC_DEFINE(HAVE_EXPAT_H)],
+		 [AC_MSG_ERROR([Can't find expat.h. Please install expat.])])
+AC_CHECK_LIB(expat, XML_ParserCreate, [EXPAT_LIBS=&quot;-lexpat&quot;],
+	     [AC_MSG_ERROR([Can't find expat library. Please install expat.])])
+AC_SUBST(EXPAT_LIBS)
 
 AC_PATH_PROG(XSLTPROC, xsltproc)
 AM_CONDITIONAL([HAVE_XSLTPROC], [test &quot;x$XSLTPROC&quot; != &quot;x&quot;])
-- 
1.9.2

</PRE>



















































<!--endarticle-->
    <HR>
    <P><UL>
        <!--threads-->
	<LI>Previous message: <A HREF="014887.html">[PATCH v4] configure.ac: Change in build system to use the path from pkg-config for wayland-scanner.
</A></li>
	<LI>Next message: <A HREF="014893.html">[PATCH 2/5] Remove wayland_protocoldir definition
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#14892">[ date ]</a>
              <a href="thread.html#14892">[ thread ]</a>
              <a href="subject.html#14892">[ subject ]</a>
              <a href="author.html#14892">[ author ]</a>
         </LI>
       </UL>

<hr>
<a href="http://lists.freedesktop.org/mailman/listinfo/wayland-devel">More information about the wayland-devel
mailing list</a><br>
</body></html>
